### PR TITLE
Fix/poseidon 1 input

### PR DESCRIPTION
**Pull request by HSG88**
Add support to Poseidon inputs of size 1 to 16

### CODE REVIEW

File: std/hash/poseidon/poseidon.go [hunk 1]
```diff
@@ -77,7 +76,7 @@ func permutation(api frontend.API, state []frontend.Variable) []frontend.Variabl
 func Poseidon(api frontend.API, input ...frontend.Variable) frontend.Variable {
 	inputLength := len(input)
 	// No support for hashing inputs of length less than 2
```
Comment (lightning-li) [inline std/hash/poseidon/poseidon.go]: This comment should be revised

**Review by lightning-li — APPROVED** [review]

File: std/hash/poseidon/poseidon.go (hunk 1 above)
Comment (lightning-li) [reply std/hash/poseidon/poseidon.go]: @HSG88 There is a failed test case "TestPoseidon256".

**Review by ZhAnGeek — APPROVED** [review]
lgtm

File: std/hash/poseidon/poseidon.go (hunk 1 above)
Comment (HSG88) [reply std/hash/poseidon/poseidon.go]: I resolved the above comments in the new commit

**Comment by lightning-li** [comment]
LGTM

**Pull request merged**